Iteration 2 con

cmougios-ITC205 opened the pull request:
Please review

cmougios-ITC205 committed:
Refactored private variables as per code style guides

cmougios-ITC205 committed:
Inserted keyword 'this' to reference current objects

cmougios-ITC205 committed:
Changed variable name ui->UI as per code style guidelines

cmougios-ITC205 committed:
Refactored method names as per the code style guidelines

cmougios-ITC205 committed:
Changed parameter-list MEMMER_ID -> memberId

cmougios-ITC205 committed:
Refactored variable name 'B' -> 'DisplayBook'

cmougios-ITC205 committed:
Refactored method variables to a more meaningful name

cmougios-ITC205 committed:
Refactored variable names to camelBack style as per the code style guidelines

cmougios-ITC205 committed:
fixed bracket style to end-of-line style as per code style guildelines

cmougios-ITC205 committed:
Refactored variable names to more meaningful names

cmougios-ITC205 committed:
Inserted keyword 'this' to reference current objects

cmougios-ITC205 committed:
Refactored variable name Member_ID -> member_Id

cmougios-ITC205 committed:
Changed indentation to 4 spaces

cmougios-ITC205 committed:
Refactored enum UiState -> UIState

cmougios-ITC205 committed:
Merge branch 'master' into iteration-2-con
Merge requested by mediator

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+    private enum ControlState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+    private ControlState state;
+
+    private List<book> pending;
```
D1: use a name more meaningful than 'pending'

cmougios-ITC205 replied:
replaced pending -> borrowBookStatus

HenleyCH replied:
D1.1: After investigation, 'borrowBookStatus' is not an appropriate name for the list. 
The list is used to hold scanned books pending completion of the borrow. 
I recommend 'booksPendingBorrow'.

cmougios-ITC205 replied:
udpated

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+    private ControlState state;
+
+    private List<book> pending;
+    private List<loan> completed;
```
D2: use a name more meaningful than 'completed'

cmougios-ITC205 replied:
replaced completed -> borrowBookCompleted

HenleyCH replied:
D2.1: After investigation, 'borrowBookCompleted' is not an appropriate name for the list.
The list is used to hold books successfully borrowed (from one use case) as loan items.
I recommend 'booksLoaned' or similar.

cmougios-ITC205 replied:
done

HenleyCH commented on BorrowBookUI.java:
```diff
@@ -2,116 +2,116 @@
+    public static enum UIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+
+    private BorrowBookControl control;
+    private Scanner input;
```
D3: use a name more meaningful than 'input'
(HINT: 'systemInInput' is fine, it tells the reader what the Scanner is scanning)

cmougios-ITC205 replied:
updated Scanner to provided HINT

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+    }
+
+
+    public void swiped(int memberId) {
```
D4: (Rule 02) use a VERB for method names, ensure it is meaningful

cmougios-ITC205 replied:
updated to memberCardSwiped

HenleyCH replied:
D4.1: please use the verbNoun format to be clear. Method name 'memberCardSwiped' could be mistaken as a boolean variable.
e.g. swipeMemberCard

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+            this.UI.setState(BorrowBookUI.UIState.RESTRICTED); }}
+
+
+    public void scanned(int bookId) {
```
D5: (Rule 02) use a VERB for method names, ensure it is meaningful

cmougios-ITC205 replied:
updated from scanned -> bookScanned

HenleyCH replied:
D5.1: please use the verbNoun format to be clear. Method 'bookScanned' could be mistaken as a variable.
e.g. scanBook

cmougios-ITC205 replied:
udpated

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+            return;
+        }
+        this.pending.add(this.book);
+        for (book displayBook : this.pending) {
```
D6: 'displayBook' could be mistaken as a verb/method, ensure it will be interpreted as a noun, ensure it is meaningful

cmougios-ITC205 replied:
updated to borrowBookList

cmougios-ITC205 replied:
updated

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+        else {
+            this.UI.display("\nFinal Borrowing List");
+            for (book displayBook : this.pending) {
+                this.UI.display(displayBook.toString());
```
D7: correct name of temporary variable 'displayBook' as described in D6 (must be a noun, must be meaningful)

cmougios-ITC205 replied:
updated

cmougios-ITC205 replied:
updated

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+    }
+
+
+    public void complete() {
```
D8: use a method name more meaningful than 'complete'

cmougios-ITC205 replied:
updated to borrowBookComplete

HenleyCH replied:
D8.1: please use the verbNoun format to be clear. Method 'borrowBookComplete' could be mistaken as a boolean variable.
e.g. completeBorrow

cmougios-ITC205 replied:
done

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+        if (!this.state.equals(ControlState.FINALISING)) {
+            throw new RuntimeException("BorrowBookControl: cannot call commitLoans except in FINALISING state");
+        }
+        for (book displayBook : this.pending) {
```
D9: (reference D6 and D7) temporary variable 'displayBook' must be a noun and must be meaningful

cmougios-ITC205 replied:
updated

HenleyCH commented on BorrowBookUI.java:
```diff
@@ -2,116 +2,116 @@
+    }
+
+
+    private String input(String prompt) {
```
D10: use a method name more meaningful than 'input'. 
e.g. 'askQuestion'

cmougios-ITC205 replied:
updated as per example

HenleyCH commented on BorrowBookUI.java:
```diff
@@ -2,116 +2,116 @@
+
+
+            case SCANNING:
+                String scanBook = input("Scan Book (<enter> completes): ");
```
D11: ensure the variable name is a NOUN 
e.g. 'scannedBook'

cmougios-ITC205 replied:
updated as per example

HenleyCH reviewed (changes requested):
Great work.
The only changes required are updates to correct method and variable names to be more meaningful and/or to ensure they are appropriately interpreted as either a verb or a noun.

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+            this.UI.setState(BorrowBookUI.UIState.SCANNING);
+            this.state = ControlState.SCANNING; }
+        else
+        {
```
D12: use end-of-line bracketing style
i.e 'else {'

cmougios-ITC205 replied:
fixed

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,116 @@
+        else
+        {
+            this.UI.display("Member cannot borrow at this time");
+            this.UI.setState(BorrowBookUI.UIState.RESTRICTED); }}
```
D13: refer to correct bracketing style. final '}' should be on next line.
Please look over other lines in both classes to ensure they also meet this standard.

cmougios-ITC205 replied:
interesting, the opening bracket shows correctly placed but the closing incorrectly in IDE. Fixed closing bracket

cmougios-ITC205 committed:
D1: Changed array list variable to a more meaningful name pending -> borrowBookStatus

D2: Changed array list variable to a more meaningful name completed -> borrowBookCompleted

D3: Changed Scanner input name to a more meaningful name from input -> systemInInput

D4: Changed method name to a more meaningful name from swiped -> memberCardSwiped

D5: Changed method name to a more meaningful name from scanned -> bookScanned

D6: Changed variable from displayBook -> borrowBookList

D7: Changed variable from displayBook -> borrowBookList

D8: Changed method name to a more meaningful name from complete -> borrowBookComplete

D9: Changed variable from displayBook -> borrowBookList

D10: Change method name to a more meaningful name from input -> askQuestion

D11: Changed variable to a NOUN from scanBook->ScannedBook

D12: Fixed bracketing to end-of-line style

D13: Fixed bracketing to end-of-line style

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -41,16 +41,16 @@ public void swiped(int memberId) {
-            this.pending = new ArrayList<>();
+            this.borrowBookStatus = new ArrayList<>();
             this.UI.setState(BorrowBookUI.UIState.SCANNING);
             this.state = ControlState.SCANNING; }
```
D14: (lines 46 and 49) move each closing bracket '}' to a new line (each should have their own line) to meet code style guidelines requirements.
Check over both files to ensure this rule is consistently enforced throughout.

cmougios-ITC205 replied:
fixed

cmougios-ITC205 committed:
D14: Fixed bracketing to end-of-line style

cmougios-ITC205 committed:
change method name to a more meaningful name

cmougios-ITC205 committed:
changed method name from bookScanned -> scanBook

cmougios-ITC205 committed:
changed method name from memberCardSwiped -> swipeMemberCard

cmougios-ITC205 committed:
fixed format if/else

cmougios-ITC205 committed:
D1.1 Changed list name to -> booksPendingBorrow

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -97,7 +97,7 @@ public void commitLoans() {
         }
-        for (book borrowBookList : this.borrowBookStatus) {
+        for (book borrowBookList : this.booksPendingBorrow) {
             loan LOAN = this.library.ISSUE_LAON(borrowBookList, this.member);
```
D15: refactor variable name 'LOAN' to meet code style guidelines.

cmougios-ITC205 replied:
done

cmougios-ITC205 committed:
D2.1 Changed list to -> booksLoaned

cmougios-ITC205 committed:
D16: refactored variable LOAN -> loan

cmougios-ITC205 committed:
D15: refactored variable LOAN -> loan

HenleyCH commented on BorrowBookControl.java:
```diff
@@ -2,116 +2,118 @@
+            this.booksLoaned.add(loan);
+        }
+        this.UI.display("Completed Loan Slip");
+        for (loan LOAN : this.booksLoaned) {
```
D16: rename temporary (iterator) variable 'LOAN' to meet code style guidelines.

cmougios-ITC205 committed:
D16: rename temporary (iterator) variable 'LOAN' to meet code style guidelines

cmougios-ITC205 committed:
D15: refactored variable LOAN -> loan

cmougios-ITC205 committed:
Merge branch 'iteration-2-con' of https://github.com/cmougios-ITC205/ITC205_GitHub_Repo into iteration-2-con
merging due to issues

cmougios-ITC205 committed:
D15: Reverting - refactored variable LOAN -> loan

cmougios-ITC205 committed:
D16: rename temporary (iterator) variable 'LOAN' to meet code style guidelines

HenleyCH reviewed (approved):
All changes satisfactorily meet code guidelines.
All non-altered code either satisfactorily meets code guidelines or is not applicable to this feature development branch.
